v2/sender.go

```diff
@@ -178,9 +184,17 @@ func (d *Sender) SendMessageBatch(ctx context.Context, messages []*azservicebus.
 
 // AzSender returns the underlying azservicebus.Sender instance.
 func (d *Sender) AzSender() AzServiceBusSender {
+	d.mu.RLock()
+	defer d.mu.RUnlock()
 	return d.sbSender
 }
 
+func (d *Sender) FailOver(sender AzServiceBusSender) {
```

Comment:
Add documentation on the Failover, and the impact on the consumers of the Sender object.